src/app/auth/services/customer-auth.service.ts

```diff
@@ -0,0 +1,105 @@
+import { HttpClient, HttpErrorResponse } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { Observable, catchError, map, tap, throwError } from 'rxjs';
+import { AuthResponse, CustomerResponse } from './interfaces';
+import { SnackbarService } from '../../shared/services/mat-snackbar.service';
+
+export type CustomerRegestrationForm = {
+  email: string;
+  password: string;
+  firstName: string;
+  lastName: string;
+  dateOfBirth: string;
+  addresses: [
+    {
+      streetName: string;
+      city: string;
+      postalCode: string;
+      country: string;
+    },
+  ];
+};
+
+@Injectable({
+  providedIn: 'root',
+})
+export class AuthCustomerService {
+  private apiUrl =
+    'https://api.europe-west1.gcp.commercetools.com/ecommerce-application-rsschool-1905';
+
+  constructor(
+    private http: HttpClient,
+    private snackbarService: SnackbarService,
+  ) {}
+
+  createCustomer(body: CustomerRegestrationForm): Observable<CustomerResponse> {
+    return this.http
+      .post<CustomerResponse>(`${this.apiUrl}/customers`, body)
+      .pipe(
+        catchError((error: HttpErrorResponse) => {
+          if (error.status === 401) {
+            return this.http.post<CustomerResponse>(
+              `${this.apiUrl}/customers`,
+              body,
+            );
+          } else {
+            this.snackbarService.show(error.error.message, 'Close', 3000);
+            return throwError(() => new Error(error.error.message));
+          }
+        }),
+      );
+  }
+
+  customerLogin(email: string, password: string): Observable<CustomerResponse> {
+    return this.http
+      .post<CustomerResponse>(`${this.apiUrl}/login`, { email, password })
+      .pipe(
+        tap((response) => {
+          console.log('here is response from customerLogin');
+          console.log(response);
+          // sessionStorage.setItem('auth_token', response.access_token); // Store the token
+        }),
+        catchError((error: HttpErrorResponse) => {
+          if (error.status === 401) {
+            return this.http.post<CustomerResponse>(`${this.apiUrl}/login`, {
+              email,
+              password,
+            });
+          } else {
+            this.snackbarService.show(error.error.message, 'Close', 3000);
+            return throwError(() => new Error(error.error.message));
+          }
+        }),
+      );
+  }
+
+  tokenForAnanimus(): Observable<AuthResponse> {
+    // const myHeaders = new HttpHeaders({
```

Comment:
Remove it if you don't need it. If you want to use it further - add a comment to specify a reason